rclpy/rclpy/task.py

```diff
@@ -42,6 +44,12 @@ def __init__(self, *, executor=None):
         self._executor = None
         self._set_executor(executor)
 
+    def __del__(self):
+        if self._exception is not None:
```

Comment:
This will print even for exceptions that were retrieved, is that intended?

Reply:
That is correct. Unfortunately I don't see a way to avoid that since there is no API calling code invoked to "take" an exception.

Reply:
I added a boolean flag. When `exception()` is being called it won't be reported assuming the caller did "the right thing" already.